File: module_security/src/main/java/com/company/payroll/config/InterceptorConfig.java
package com.company.payroll.config;

import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

import com.company.payroll.interceptor.ApiInterceptor;

@Configuration
public class InterceptorConfig implements WebMvcConfigurer {

	@Override
	public void addInterceptors(InterceptorRegistry registry) {
		registry.addInterceptor(new ApiInterceptor())
		.addPathPatterns("/api/**")
		.excludePathPatterns("/api/users/login", "/logout", "/v3/api-docs/**", "/swagger-ui/**")
		.order(2);
	}
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/configuration/InterceptorConfig.kt
package com.company.payroll.configuration

import com.company.payroll.interceptor.ApiInterceptor
import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.InterceptorRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class InterceptorConfig: WebMvcConfigurer {
  override fun addInterceptors(registry: InterceptorRegistry) {
    registry.addInterceptor(ApiInterceptor())
        .addPathPatterns("/api/**")
        .excludePathPatterns("/api/users/login", "/v3/api-docs/**", "/swagger-ui/**")
        .order(2)
  }
}